stdlib/public/Darwin/Foundation/Data.swift

```diff
@@ -2388,10 +2352,14 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
             // Copy the contents of the buffer...
             _representation.append(contentsOf: UnsafeRawBufferPointer(start: base, count: endIndex))
 
-            /// ... and append the rest byte-wise.
+            // ... and append the rest byte-wise, buffering through an InlineData.
+            var buffer = InlineData()
             while let element = iter.next() {
-                Swift.withUnsafeBytes(of: element) {
-                    _representation.append(contentsOf: $0)
+                if buffer.count < buffer.capacity {
+                    buffer.append(byte: element)
+                } else {
+                    buffer.withUnsafeBytes {_representation.append(contentsOf: $0)}
+                    buffer.count = 0
                 }
             }
```

Comment:
Drain the buffer at the end?

Reply:
Same as above — `buffer.count = 0` takes care of this.